app/models/filter.rb

```diff
@@ -3,24 +3,39 @@
 # Table name: filters
 #
 #  id         :bigint           not null, primary key
+#  icon       :string
+#  is_pinned  :boolean          default(FALSE), not null
 #  name       :string           not null
 #  position   :integer          not null
-#  query      :string           not null
+#  query      :string
+#  type       :string           not null
 #  created_at :datetime         not null
 #  updated_at :datetime         not null
-#  author_id  :bigint           not null
+#  author_id  :bigint
+#  tag_id     :bigint
 #  tenant_id  :bigint           not null
 #
 class Filter < ApplicationRecord
   include AuditableEvents
+  include Iconized
 
-  belongs_to :author, class_name: 'User'
+  belongs_to :author, class_name: 'User', optional: true
   belongs_to :tenant
+  belongs_to :tag, optional: true
+  has_many :user_filter_visibilities, inverse_of: :filter, dependent: :destroy
 
-  validates :tenant_id, :author_id, :name, :query, presence: true
+  validates :tenant_id, :name, presence: true
 
   before_create :fill_position
 
+  scope :pinned, -> { where(is_pinned: true) }
+  scope :not_pinned, -> { where(is_pinned: false) }
+  scope :visible_for, -> (user) { joins(:user_filter_visibilities)
+    .where(user_filter_visibilities: { visible: true, user: user})
+    .where(is_pinned: false) }
```

Comment:
Aky vyznam ma tento atribut `is_pinned`? Ak spravne pozeram, tak defaultne je false, ale nevidim, ze by sa niekedy tato hodnota menila.

Reply:
Toto som dal prec, s Ahmedom sme sa dohodli, ze to netreba, bol to len nejaky povodny pokus o nieco.